File: JavaScript/Math-Array-String/31.js
// 1528. Shuffle String

/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    // Create an array to hold the shuffled characters
    let shuffled = new Array(s.length);
    
    // Place each character at the correct index
    for (let i = 0; i < s.length; i++) {
        shuffled[indices[i]] = s[i];
    }
    
    // Join the array into a single string and return it
    return shuffled.join('');
};

// Example usage
let s = "codeleet";
let indices = [4, 5, 6, 7, 0, 2, 1, 3];
console.log(restoreString(s, indices)); // Output: "leetcode"

File: Python/Math-Array-String/31.py
# 1528. Shuffle String

from typing import List

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        # Create a list to hold the shuffled characters
        shuffled = [''] * len(s)
        
        # Place each character at the correct index
        for i, char in enumerate(s):
            shuffled[indices[i]] = char
        
        # Join the list into a single string and return
        return ''.join(shuffled)
    
sol = Solution()
output = sol.restoreString("codeleet", [4, 5, 6, 7, 0, 2, 1, 3])
print(output)